path_login_test.go

```diff
@@ -1474,3 +1475,58 @@ func requireErrorCode(t *testing.T, err error, expectedCode int) {
 		t.Fatalf("wrong error code, expected %d, got %d", expectedCode, codedErr.Code())
 	}
 }
+
+func TestResolveRole(t *testing.T) {
+	b, storage := getBackend(t)
+	role := "testrole"
+
+	validRoleStorageEntry := &roleStorageEntry{
+		TokenParams: tokenutil.TokenParams{
+			TokenPolicies:   []string{"test"},
+			TokenPeriod:     3 * time.Second,
+			TokenTTL:        1 * time.Second,
+			TokenMaxTTL:     5 * time.Second,
+			TokenNumUses:    12,
+			TokenBoundCIDRs: nil,
+		},
+		Policies:                 []string{"test"},
+		Period:                   3 * time.Second,
+		ServiceAccountNames:      []string{"name"},
+		ServiceAccountNamespaces: []string{"namespace"},
+		TTL:                      1 * time.Second,
+		MaxTTL:                   5 * time.Second,
+		NumUses:                  12,
+		BoundCIDRs:               nil,
+		AliasNameSource:          aliasNameSourceDefault,
+	}
+
+	entry, err := logical.StorageEntryJSON("role/"+role, validRoleStorageEntry)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := storage.Put(context.TODO(), entry); err != nil {
```

Comment:
nit: functionally they're the same, but Background signals the intent better here

```suggestion
	if err := storage.Put(context.Background(), entry); err != nil {
```